users/vieting/jobs/fairseq_finetuny.py

```diff
@@ -0,0 +1,185 @@
+        """
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
+        :param str file_extension: file extension to look for in audio_dir_path
+        :param float valid_percent: percentage of files to be in validation set
+        :param int seed: random seed for splitting into train and valid set
+        :param str|None path_must_contain: if set, path must contain this substring
+            for a file to be included in the task
+        """
+        if isinstance(corpus_paths, tk.Path):
+            self.corpus_paths = [corpus_paths]
+        else:
+            assert isinstance(corpus_paths, list)
+            self.corpus_paths = corpus_paths
+        assert min([isinstance(path, tk.Path) for path in self.corpus_paths])
+
+        self.file_extension = file_extension
+        self.valid_percent = valid_percent
+        assert 0.0 <= self.valid_percent <= 1.0
+        self.seed = seed
+        self.path_must_contain = path_must_contain
+
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
+
+        self.out_task_path = self.output_path("task", directory=True)
+
+        self.out_train_tsv_path = self.output_path("task/train.tsv")
+        self.out_valid_tsv_path = self.output_path("task/valid.tsv")
+        
+        self.out_dict_ltr_path = self.output_path("task/dict.ltr.txt")
+
+        self.out_train_ltr_path = self.output_path("task/train.ltr")
+        self.out_train_wrd_path = self.output_path("task/train.wrd")
+        self.out_valid_ltr_path = self.output_path("task/valid.ltr")
+        self.out_valid_wrd_path = self.output_path("task/valid.wrd")
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        #self.create_manifest()
+        self.create_tsv_and_labels()
+        self.download_dict_ltr()
+    
+    def get_common_dir(self):
+        """
+        Returns the common directory of all audios given in the corpora.
+        """
+        # TODO test this
+        common_dir = None
+        for corpus_path in self.corpus_paths:
+            c = corpus.Corpus()
+            c.load(corpus_path.get())
+            for segment in c.segments():
+                audio_path = segment.recording.audio
+                assert os.path.exists(audio_path), f"Path {audio_path} does not exist."
+                if common_dir is None:
+                    common_dir = os.path.dirname(audio_path)
+                else:
+                    common_dir = os.path.commonpath([common_dir, os.path.dirname(audio_path)])
+        return common_dir
+
+    def create_tsv_and_labels(self):
+        """
+        Creates both tsv files (train.tsv, valid.tsv) and labels (train.ltr, train.wrd, valid.ltr, valid.wrd) from the given corpora.
+        """
+        rand = random.Random(self.seed)
+        
+        common_dir = self.get_common_dir()
+
+        valid_tsv = (
+            open(self.out_valid_tsv_path, "w")
+            if self.valid_percent > 0
```

Comment:
I don't think you need to condition it here. If `valid_percent` is 0, no files would be added. I think the difference is that in this case you would get an empty file, but I think that's intended behavior anyway because otherwise you define an output that then does not exist.